orderer/consensus/smartbft/chain.go

```diff
@@ -196,22 +196,32 @@ func NewChain(
 	heartbeatTicker := time.NewTicker(1 * time.Second)
 	heartbeatTimeout := 10 * time.Second
 	heartbeatCount := uint64(5)
-	var role Role
-	if selfID%2 == 0 {
-		role = HeartbeatSender
-	} else {
-		role = HeartbeatReceiver
-	}
-	senders := make([]uint64, 0)
-	receivers := make([]uint64, 0)
+
+	myRole := HeartbeatSender
+
+	heartbeatReceivers := make([]uint64, 0)
 	for _, node := range currentCommittee {
-		if node.ID%2 == 0 {
-			senders = append(senders, uint64(node.ID))
-		} else {
-			receivers = append(receivers, uint64(node.ID))
+		if selfID == uint64(node.ID) {
+			myRole = HeartbeatReceiver
+		}
+		heartbeatReceivers = append(heartbeatReceivers, uint64(node.ID))
+	}
+
+	heartbeatSenders := make([]uint64, 0)
+	for _, node := range rtc.Nodes {
+		receiver := false
+		for _, committeeNode := range currentCommittee {
+			if node == uint64(committeeNode.ID) {
+				receiver = true
+				break
+			}
+		}
+		if receiver == false {
+			heartbeatSenders = append(heartbeatSenders, node)
 		}
 	}
-	c.heartbeatMonitor = NewHeartbeatMonitor(c.consensus.Comm.(MessageSender), heartbeatTicker.C, logger, heartbeatTimeout, heartbeatCount, role, senders, receivers)
+
+	c.heartbeatMonitor = NewHeartbeatMonitor(c.consensus.Comm.(MessageSender), heartbeatTicker.C, logger, heartbeatTimeout, heartbeatCount, myRole, heartbeatSenders, heartbeatReceivers)
```

Comment:
when we change a committee upon a block commit, we need to re-create this monitor.

Reply:
true
what should I do about this now?

Reply:
I suggest you hold a reference to the monitor and upon every block commit, when we compute the new committee, you will make the required adjustments according to whether there was a committee change, or not.